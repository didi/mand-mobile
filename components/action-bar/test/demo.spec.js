import Demo0 from './cases/demo0'
import Demo1 from './cases/demo1'
import Demo2 from './cases/demo2'
import Demo3 from './cases/demo3'
import {renderToString} from '@vue/server-test-utils'

describe('ActionBar', () => {
  test(`Basic`, () => {
    expect(renderToString(Demo0)).toMatchSnapshot()
  })
  test(`Multiple buttons`, () => {
    expect(renderToString(Demo1)).toMatchSnapshot()
  })
  test(`Multiple buttons and disabled button`, () => {
    expect(renderToString(Demo2)).toMatchSnapshot()
  })
  test(`With slot`, () => {
    expect(renderToString(Demo3)).toMatchSnapshot()
  })
})
