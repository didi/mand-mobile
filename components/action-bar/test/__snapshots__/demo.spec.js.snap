// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActionBar Demo Basic 1`] = `
<div class="md-example-child md-example-child-action-bar md-example-child-0">
  <div class="md-action-bar">
    <div class="md-action-bar-container">
      <!---->
      <div class="md-action-bar-group"><button type="button" class="md-button md-action-bar-button primary active block">
          <div class="md-button-inner">
            <!---->
            <div class="md-button-content">
              主要按钮
            </div>
          </div>
        </button></div>
    </div>
  </div>
</div>
`;

exports[`ActionBar Demo Multiple buttons 1`] = `
<div class="md-example-child md-example-child-action-bar md-example-child-0">
  <div class="md-action-bar">
    <div class="md-action-bar-container">
      <!---->
      <div class="md-action-bar-group"><button type="button" class="md-button md-action-bar-button primary active block plain">
          <div class="md-button-inner">
            <!---->
            <div class="md-button-content">
              次要按钮
            </div>
          </div>
        </button><button type="button" class="md-button md-action-bar-button primary active block">
          <div class="md-button-inner">
            <!---->
            <div class="md-button-content">
              主要按钮
            </div>
          </div>
        </button></div>
    </div>
  </div>
</div>
`;

exports[`ActionBar Demo Multiple buttons and disabled button 1`] = `
<div class="md-example-child md-example-child-action-bar md-example-child-1">
  <div class="md-action-bar">
    <div class="md-action-bar-container">
      <!---->
      <div class="md-action-bar-group"><button type="button" disabled="disabled" class="md-button md-action-bar-button disabled active block plain">
          <div class="md-button-inner">
            <!---->
            <div class="md-button-content">
              次要按钮
            </div>
          </div>
        </button><button type="button" disabled="disabled" class="md-button md-action-bar-button disabled active block">
          <div class="md-button-inner">
            <!---->
            <div class="md-button-content">
              主要按钮
            </div>
          </div>
        </button></div>
    </div>
  </div>
</div>
`;

exports[`ActionBar Demo With slot 1`] = `
<div class="md-example-child md-example-child-action-bar md-example-child-3">
  <div class="md-action-bar">
    <div class="md-action-bar-container">
      <div class="md-action-bar-text"><span class="price">¥128.00<small>起</small></span></div>
      <div class="md-action-bar-group"><button type="button" class="md-button md-action-bar-button primary active block">
          <div class="md-button-inner">
            <!---->
            <div class="md-button-content">
              主要按钮
            </div>
          </div>
        </button></div>
    </div>
  </div>
</div>
`;
